docs/HOL_Automated-Testing/HOL_Automated-Testing.md

```diff
@@ -0,0 +1,226 @@
+
+**18.** Follow **Step 12** again to create a **new Java Project**.
+
+**19.** In the New Java Project dialog name the project **Clients**, uncheck the **Use default location**, choose the **Clients** folder located in `src/` of the PartsUnlimitedMRP repository, and click **Finish**.
+
+![](media/new_clients.png)
+
+**20.** In the Package Explorer window, **right-click on Clients** and then select **Configure -> Convert to Gradle Project**.
+
+You should now have 3 projects in Eclipse that are all Gradle projects (signified by the `G` in the glyph of the projects in the Package Explorer window).
+
+![](media/gradle_projects.png)
+
+**21.** In Eclipse's menu select **Window -> Show View -> Other...**
+
+![](media/other_view.png)
+
+**22.** Type `gradle` in the filter box, and click **OK**.
+
+You can now select one of the three projects that were created, and be able to view all of the Gradle tasks in each on via the Gradle Tasks window.
+
+![](media/order_gradle_tasks.png)
+
+The development environment is now set up, and you should be ready to make write some automated tests. Go ahead and look over the different tasks - you may even want to try out the `build` task.
+
+###2. Add JUnit tests to Parts Unlimited MRP App###
+
+This task will focus on creating a unit test to test a part of the Parts Unlimited App.
+
+**1.** There are already a bunch of JUnit tests that exist within the various different projects of the Parts Unlimited MRP App. **Open up the file** in the **OrderService** Eclipse project located here:
+
+     src/test/java/smpl.ordering.controllers/CatalogControllerTest.java
+
+![](media/open_catalog_tests.png)
+
+**2.** Lets go ahead and run these tests. In the menu select **Run -> Run As -> JUnit Test**.
+
+![](media/run_default_tests.png)
+
+This should open up the JUnit window and give you a summary of the test run.
+
+![](media/default_test_run.png)
+
+**3.** Paste the following code into the `CatalogControllerTest.java` file:
+
+	@Test
+	public void testAddMultipleCatalogItems() throws Exception
+	{
+    	ResponseEntity<List<CatalogItem>> oldItems = controller.getCatalogItems();
+    	int sizeBeforeAdds = 0;
+    	if (oldItems.getBody() != null)
+    	{
+        	sizeBeforeAdds = oldItems.getBody().size();
+    	}
+    	
+    	ResponseEntity<CatalogItem> response = 
+    			controller.addCatalogItem(new CatalogItem("ACC-0123", "Storage", 15.75, 5, 2));
+    	assertNotNull(response);
+    	assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    	
+    	response = 
+    			controller.addCatalogItem(new CatalogItem("ACC-0124", "Storage", 11.25, 7, 6));
+    	assertNotNull(response);
+    	assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    	
+    	ResponseEntity<List<CatalogItem>> newItems = controller.getCatalogItems();
+    	int sizeAfterAdds = newItems.getBody().size();
+    	
+    	assertEquals(sizeBeforeAdds + 2, sizeAfterAdds);
+	}
+
+The final result should look something like this:
```

Comment:
Screenshot does not match adding "testAddMultipleCatalogItems()". Instead screenshow shows adding "testAddCatalogItems()".